pkg/controller/common/errors.go

```diff
@@ -0,0 +1,54 @@
+package common
+
+import (
+	"github.com/go-logr/logr"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// NonRetriableCtrlError wraps errors with the addition of having
+// the information for the error being non-retriable
+type NonRetriableCtrlError struct {
```

Comment:
Because everyone's favourite topic in software is naming things...I'm not super fond of `NonRetriableCtrlError`. It really sounds to me like the error can't ever be retried. Would something neutral like `ComplianceCtrlError` work?

Sorry for nitpicking.

Reply:
btw all the method names sound good to me.

Reply:
That was exactly the intent though. The error is non reyriable. And I only wrap errors that can't be retried.

Reply:
Yeah, we talked on Slack about how you switched the logic to signaling that some errors can't be retried and I still think it's a good change. So with the naming you meant "Take an error and wrap it as non-retriable?" That makes more sense and also makes sense together with `IsRetriable` returning true on any other type than `NonRetriableCtrlError`.

Now I /really/ with we could use the new error mechanism because the code would be clearer if it said:
```
if errors.Is(err, NonRetriableCtrlError)  {
 // fail hard
}
```

Reply:
True... But I didn't figure out how to use an image with go 1.13. At least not with go-toolset. Maybe if we switched to Fedora for the build image.